### PR TITLE
mvcc: skip accumulated locks by a second get

### PR DESCRIPTION
**Pull request by sticnarf**
### What is changed and how it works?

Issue Number: Ref #13694

What's Changed:

```commit-message
Prewrite and acquire_pessimistic_lock use get_write to get the latest PUT or
DELETE record. Point get is implemented by PointGetter. In these cases, with
last_change_ts and versions_to_last_change, now we use an additional get
operation to find the record directly if versions_to_last_change reaches
SEEK_BOUND.

I think no additional metrics are needed because the count of get operation
of the write CF represents represents it.
```

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by sticnarf** [comment]
@MyonKeminta Whether do you think `SEEK_BOUND` is a good threshold for skipping? I find you did some research about this bound years ago.

**Comment by MyonKeminta** [comment]
> @MyonKeminta Whether do you think `SEEK_BOUND` is a good threshold for skipping? I find you did some research about this bound years ago.

It makes sense to me to do the `get_cf` directly (which seems to include a `seek` operation (?))  if the accumulated locks/rollbacks is more than `SEEK_BOUND`. `SEEK_BOUND` is expected to stand for how many `next`s' overhead is as large as a `seek`'s. But since that research were done years ago, I'm actually not sure if its value is still suitable nowadays as TiKV and rocksdb are both evolving.

**Comment by sticnarf** [comment]
@MyonKeminta @ekexium PTAL

File: src/storage/mvcc/reader/reader.rs [hunk 1]
```diff
@@ -382,7 +382,31 @@ impl<S: EngineSnapshot> MvccReader<S> {
...
+                            if write.versions_to_last_change < SEEK_BOUND
+                                || write.last_change_ts.is_zero()
+                            {
+                                ts = commit_ts.prev();
+                            } else {
+                                let commit_ts = write.last_change_ts;
+                                let key_with_ts = key.clone().append_ts(commit_ts);
+                                let Some(value) = self
+                                    .snapshot
+                                    .get_cf(CF_WRITE, &key_with_ts)? else {
```
Comment (ekexium) [inline src/storage/mvcc/reader/reader.rs:395]: Wow

**Review by ekexium — APPROVED** [review]

File: src/storage/mvcc/reader/reader.rs (hunk 1 above)
Comment (MyonKeminta) [reply src/storage/mvcc/reader/reader.rs:395]: Wow

**Review by MyonKeminta — APPROVED** [review]

**Pull request merged**